go_game/gameLogic.ts

```diff
@@ -27,15 +47,45 @@ type Sets = {white: Points[]; black: Points[]};
 export function isPosOnHintLine(row: number, col: number, hint: RiddleData) {
   switch (hint) {
     case 'r1':
-      return row == 0;
+      return row == 0 && col == 4;
     case 'r2':
-      return row == 1;
+      return row == 1 && col == 5;
```

Comment:
This is repeating the same exact logic as riddleHints (which is pointless).
You can simply do:

export function isPosOnHintLine(row: number, col: number, hint: RiddleData) {
  const h = riddleHints(hint);
  return row == h.row && col == h.col;
}